### PR TITLE
added faq dropdown component

### PR DESCRIPTION
**Pull request by agonzalez22**
Added the dropdown component with spinning animations for icons
image
image

### CODE REVIEW

File: packages/ui/src/FAQDropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+export default function FAQDropdown({
+  dropdownQuestion,
+  dropdownAnswer,
+}: FAQDropdownProps): React.ReactNode {
+  const [isOpen, setIsOpen] = useState(false);
+
+  const toggleAccordion = () => {
+    setIsOpen(!isOpen);
```
Comment (nelson-dong) [inline packages/ui/src/FAQDropdown.tsx:19]: for better code practices use the callback for the state change
```setIsOpen((prevIsOpen) => !prevIsOpen);```

there is some weird state stuff with react where sometimes the state of isOpen is not always up to date at the right time. using the callback version ensures that you always have the right state.

**Review by nelson-dong — CHANGES_REQUESTED** [review]
image
 text explodes when too long!

File: packages/ui/package.json [hunk 2]
```diff
@@ -32,5 +32,12 @@
     "eslint": "^9.14.0",
     "postcss": "^8.4.35",
     "tailwindcss": "^3.4.1"
+  },
+  "dependencies": {
+    "@emotion/react": "^11.13.5",
```
Comment (nelson-dong) [inline packages/ui/package.json:37]: not sure but are all of these dependencies for MUI neccessary? Looks like you are only using it for the icons right now.

Comment (NelsonDong41) [reply packages/ui/package.json:37]: https://heroicons.com/

File: apps/main/src/app/(landing)/page.tsx [hunk 3]
```diff
@@ -1,13 +1,18 @@
 import { Section } from "@repo/ui";
 import { Stats } from "./Sections";
+import { FAQDropdown } from "@repo/ui";
 import { Footer, NavBar } from "../lib/Components";
 import React from "react";
 
 export default function Page(): JSX.Element {
+  const q = "Question?";
+  const a =
+    "ANswejoidfjlsdkjfoweidjlksdfjklsdjflskjhfgksjdhglskdjfghslkdjfghslkdfjghskldjfghskldjfghsdkljfghsdlkfg";
```
Comment (NelsonDong41) [inline apps/main/src/app/(landing)/page.tsx:10]: remove before merge!

File: packages/ui/package.json [hunk 4]
```diff
@@ -32,5 +32,12 @@
...
+    "@emotion/styled": "^11.13.5",
+    "@mui/icons-material": "^6.1.8",
+    "@mui/material": "^6.1.8",
+    "@mui/utils": "^6.1.8"
   }
```
Comment (NelsonDong41) [inline packages/ui/package.json:42]: are these used now that icons are from tailwind

File: packages/ui/src/FAQDropdown.tsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+      />
+    </svg>
+  );
+
+  const toggleAccordion = () => {
+    setIsOpen((prevIsOpen) => !prevIsOpen);
+  };
+
+  return (
+    <>
```
Comment (NelsonDong41) [inline packages/ui/src/FAQDropdown.tsx:51]: no need for wrapper fragment, div at top level wraps everything

**Review by NelsonDong41 — CHANGES_REQUESTED** [review]

File: packages/ui/package.json [hunk 6]
```diff
@@ -32,5 +32,6 @@
     "eslint": "^9.14.0",
     "postcss": "^8.4.35",
     "tailwindcss": "^3.4.1"
-  }
+  },
+  "dependencies": {}
```
Comment (NelsonDong41) [inline packages/ui/package.json:36]: ???!!

File: yarn.lock [hunk 7]
```diff
@@ -128,6 +128,13 @@
   dependencies:
     "@babel/types" "^7.26.0"
 
+"@babel/runtime@^7.12.5", "@babel/runtime@^7.18.3", "@babel/runtime@^7.26.0", "@babel/runtime@^7.5.5", "@babel/runtime@^7.8.7":
+  version "7.26.0"
+  resolved "https://registry.yarnpkg.com/@babel/runtime/-/runtime-7.26.0.tgz#8600c2f595f277c60815256418b85356a65173c1"
+  integrity sha512-FDSOghenHTiToteC/QRlv2q3DhPZ/oOXTBoirfWNx1Cx3TMVcGWQtMMmQcSvb/JjpNeGzx8Pq/b4fKEJuWm1sw==
+  dependencies:
+    regenerator-runtime "^0.14.0"
```
Comment (NelsonDong41) [inline yarn.lock:136]: yarn

**Review by NelsonDong41 — APPROVED** [review]

**Pull request merged**